### PR TITLE
fix issue-10446, add test case for ebs_config.volumes_per_instance check

### LINKED ISSUE #10446: EMR recreates resources when `ebs_config.volumes_per_instance` is greater than 1
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.13

### Affected Resource(s)

* `aws_emr_cluster`

### Terraform Configuration Files

```hcl
resource "aws_emr_cluster" "emr_cluster" {
  name          = "my-cluster"
  release_label = "emr-5.26.0"
  applications  = ["Hadoop", "Hive", "Pig", "Hue", "Spark"]

  ebs_root_volume_size = 10

  termination_protection            = false
  keep_job_flow_alive_when_no_steps = true
  scale_down_behavior               = "TERMINATE_AT_TASK_COMPLETION"

  master_instance_group {
    instance_type = "m5.xlarge"

    ebs_config {
      size                 = 32
      type                 = "gp2"
      volumes_per_instance = 2
    }
  }

  core_instance_group {
    instance_type  = "m5.xlarge"
    instance_count = 2

    ebs_config {
      size                 = 128
      type                 = "gp2"
      volumes_per_instance = 2
    }
  }
}
```

### Expected Behavior

Subsequent `terraform apply` calls should not recreate the cluster when the resource hasn't changed.

### Actual Behavior

The cluster is recreated every time a `terraform apply` is run.

```
-/+ aws_emr_cluster.emr_cluster (new resource required)
...snip...
      core_instance_group.0.ebs_config.2569541827.iops:                   "0" => "0"
      core_instance_group.0.ebs_config.2569541827.size:                   "128" => "0" (forces new resource)
      core_instance_group.0.ebs_config.2569541827.type:                   "gp2" => "" (forces new resource)
      core_instance_group.0.ebs_config.2569541827.volumes_per_instance:   "1" => "0" (forces new resource)
      core_instance_group.0.ebs_config.2986691328.iops:                   "" => ""
      core_instance_group.0.ebs_config.2986691328.size:                   "" => "128" (forces new resource)
      core_instance_group.0.ebs_config.2986691328.type:                   "" => "gp2" (forces new resource)
      core_instance_group.0.ebs_config.2986691328.volumes_per_instance:   "" => "2" (forces new resource)
      core_instance_group.0.id:                                           "ig-1MCB05JHBWE0E" => <computed>
      core_instance_group.0.instance_count:                               "2" => "2"
      core_instance_group.0.instance_type:                                "m5.xlarge" => "m5.xlarge"
      core_instance_type:                                                 "m5.xlarge" => <computed>
...snip...
      master_instance_group.0.ebs_config.2636219798.iops:                 "0" => "0"
      master_instance_group.0.ebs_config.2636219798.size:                 "32" => "0" (forces new resource)
      master_instance_group.0.ebs_config.2636219798.type:                 "gp2" => "" (forces new resource)
      master_instance_group.0.ebs_config.2636219798.volumes_per_instance: "1" => "0" (forces new resource)
      master_instance_group.0.ebs_config.3054294613.iops:                 "" => ""
      master_instance_group.0.ebs_config.3054294613.size:                 "" => "32" (forces new resource)
      master_instance_group.0.ebs_config.3054294613.type:                 "" => "gp2" (forces new resource)
      master_instance_group.0.ebs_config.3054294613.volumes_per_instance: "" => "2" (forces new resource)
      master_instance_group.0.id:                                         "ig-B0L2KTB8QGNY" => <computed>
      master_instance_group.0.instance_count:                             "1" => "1"
      master_instance_group.0.instance_type:                              "m5.xlarge" => "m5.xlarge"
...snip...
```

### Steps to Reproduce

1. `terraform apply`
2. `terraform apply`

### References

* Maybe related to #5111, #5075
* Terraform docs: https://www.terraform.io/docs/providers/aws/r/emr_cluster.html#ebs_config-3
* EMR docs for [describe-cluster](https://docs.aws.amazon.com/emr/latest/APIReference/API_DescribeCluster.html) and [list-instances](https://docs.aws.amazon.com/emr/latest/APIReference/API_ListInstances.html)

### PR DESCRIPTION
**Pull request by c4po**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #10446 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
BUG FIXES

* resource/aws_emr_cluster: Prevent EMR cluster recreate when `ebs_config.volumes_per_instance` is greater than 1
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSEMRCluster_ebs_config'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEMRCluster_ebs_config -timeout 120m
=== RUN   TestAccAWSEMRCluster_ebs_config
=== PAUSE TestAccAWSEMRCluster_ebs_config
=== CONT  TestAccAWSEMRCluster_ebs_config
2020/08/26 17:29:07 [INFO] Test: Using us-east-1 as test region
2020/08/26 17:29:07 [INFO] AWS Auth provider used: "EnvProvider"
--- PASS: TestAccAWSEMRCluster_ebs_config (576.89s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	576.948s
```

### CODE REVIEW

File: aws/resource_aws_emr_cluster.go [hunk 1]
```diff
@@ -1424,8 +1424,7 @@ func flattenEBSConfig(ebsBlockDevices []*emr.EbsBlockDevice) *schema.Set {
 		if ebs.VolumeSpecification.VolumeType != nil {
 			ebsAttrs["type"] = *ebs.VolumeSpecification.VolumeType
 		}
-		ebsAttrs["volumes_per_instance"] = 1
-
+		ebsAttrs["volumes_per_instance"] = len(ebsBlockDevices)
```
Comment (bflad) [inline aws/resource_aws_emr_cluster.go:1427]: Is it possible to specify multiple `ebs_config` (multiple configurations with 1 or more volumes per instance)? e.g.

```terraform
  master_instance_group {
    instance_type = "m4.large"

    ebs_config {
      size                 = 32
      type                 = "gp2"
      volumes_per_instance = 2
    }

    ebs_config {
      size                 = 50
      type                 = "gp2"
      volumes_per_instance = 2
    }
  }
```

If so, the `volumes_per_instance` needs to be calculated per configurations type.

Comment (c4po) [reply aws/resource_aws_emr_cluster.go:1427]: I don't think this is an issue here. The `volumes_per_instance` is an attribute of `ebs_config`.

Comment (c4po) [reply aws/resource_aws_emr_cluster.go]: Ok, I get what you're saying.
I updated the PR and test case
but we can not dealing with 2 same ebs_config config like this
```
  master_instance_group {
    instance_type = "m4.large"

    ebs_config {
      size                 = 32
      type                 = "gp2"
      volumes_per_instance = 2
    }

    ebs_config {
      size                 = 32
      type                 = "gp2"
      volumes_per_instance = 1
    }
  }
```

Comment (bflad) [reply aws/resource_aws_emr_cluster.go]: I think not being able to handle that last configuration there is okay -- seems reasonable to say that should have one `ebs_config` with `volumes_per_instance = 3` 👍  Also thank you for adding the additional configuration to the test case!

**Pull request merged**